### PR TITLE
python310Packages.wsgiproxy2: 0.4.2 -> 0.5.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
https://github.com/gawel/WSGIProxy2/blob/master/CHANGES.rst#051-2021-08-26

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 175281` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>libceph</li>
    <li>python310Packages.cypherpunkpay</li>
    <li>python310Packages.pyramid_chameleon</li>
    <li>python310Packages.pyramid_hawkauth</li>
    <li>python310Packages.pyramid_jinja2</li>
    <li>python39Packages.cypherpunkpay</li>
    <li>python39Packages.pyramid_chameleon</li>
    <li>python39Packages.pyramid_hawkauth</li>
    <li>python39Packages.pyramid_jinja2</li>
    <li>qemu_full</li>
    <li>samba4Full</li>
  </ul>
</details>
<details>
  <summary>46 packages built:</summary>
  <ul>
    <li>ankisyncd</li>
    <li>buildbot (python39Packages.buildbot)</li>
    <li>buildbot-full (python39Packages.buildbot-full)</li>
    <li>buildbot-ui (python39Packages.buildbot-ui)</li>
    <li>devpi-client</li>
    <li>devpi-server</li>
    <li>errbot</li>
    <li>pypiserver (python39Packages.pypiserver)</li>
    <li>python310Packages.aiohttp-apispec</li>
    <li>python310Packages.beaker</li>
    <li>python310Packages.buildbot</li>
    <li>python310Packages.buildbot-full</li>
    <li>python310Packages.buildbot-ui</li>
    <li>python310Packages.cornice</li>
    <li>python310Packages.deform</li>
    <li>python310Packages.nplusone</li>
    <li>python310Packages.pecan</li>
    <li>python310Packages.pypiserver</li>
    <li>python310Packages.pypugjs</li>
    <li>python310Packages.pyramid</li>
    <li>python310Packages.pyramid_beaker</li>
    <li>python310Packages.pyramid_exclog</li>
    <li>python310Packages.pyramid_mako</li>
    <li>python310Packages.pyramid_multiauth</li>
    <li>python310Packages.webargs</li>
    <li>python310Packages.webtest</li>
    <li>python310Packages.webtest-aiohttp</li>
    <li>python310Packages.wsgiproxy2</li>
    <li>python39Packages.aiohttp-apispec</li>
    <li>python39Packages.beaker</li>
    <li>python39Packages.cornice</li>
    <li>python39Packages.deform</li>
    <li>python39Packages.nplusone</li>
    <li>python39Packages.pecan</li>
    <li>python39Packages.pypugjs</li>
    <li>python39Packages.pyramid</li>
    <li>python39Packages.pyramid_beaker</li>
    <li>python39Packages.pyramid_exclog</li>
    <li>python39Packages.pyramid_mako</li>
    <li>python39Packages.pyramid_multiauth</li>
    <li>python39Packages.webargs</li>
    <li>python39Packages.webtest</li>
    <li>python39Packages.webtest-aiohttp</li>
    <li>python39Packages.wsgiproxy2</li>
    <li>tribler</li>
    <li>zerobin</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Looks like that `ceph` is broken on `master`.

```bash
   46 |     IOOptions io_opts;
      |               ^~~~~~~
[100%] Linking CXX static library librocksdb.a
[100%] Built target rocksdb
[ 25%] Performing install step for 'rocksdb_ext'
[ 25%] Completed 'rocksdb_ext'
[ 25%] Built target rocksdb_ext
make: *** [Makefile:146: all] Error 2
error: builder for '/nix/store/3r9k3a8y1c03ylgy8a6fz9gw7hx00ylh-ceph-16.2.7.drv' failed with exit code 2;
       last 10 log lines:
       > /build/ceph-16.2.7/src/rocksdb/env/composite_env_wrapper.h:48:25: note: by argument 3 of type 'const rocksdb::IOOptions&' to 'rocksdb::IOStatus (rocksdb::FSSequentialFile::)(size_t, const rocksdb::IOOptions&, rocksdb::Slice*, char*, rocksdb::IODebugContext*)' {aka 'rocksdb::IOStatus (rocksdb::FSSequentialFile::)(long unsigned int, const rocksdb::IOOptions&, rocksdb::Slice*, char*, rocksdb::IODebugContext*)'}
       > /build/ceph-16.2.7/src/rocksdb/env/composite_env_wrapper.h:46:15: note: 'io_opts' declared here
       >    46 |     IOOptions io_opts;
       >       |               ^~~~~~~
       > [100%] Linking CXX static library librocksdb.a
       > [100%] Built target rocksdb
       > [ 25%] Performing install step for 'rocksdb_ext'
       > [ 25%] Completed 'rocksdb_ext'
       > [ 25%] Built target rocksdb_ext
       > make: *** [Makefile:146: all] Error 2
       For full logs, run 'nix log /nix/store/3r9k3a8y1c03ylgy8a6fz9gw7hx00ylh-ceph-16.2.7.drv'.
```

**Comment by fabaff** [comment]
- `pyramid_chameleon` fixed in #175499

File: pkgs/development/python-modules/wsgiproxy2/default.nix [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
+    hash = "sha256-ouofw3cBQzBwSh3Pdtdl7KI2pg/T/z3qoh8zoeiKiSs=";
   };
 
-  propagatedBuildInputs = [ six webob ];
+  propagatedBuildInputs = [
+    webob
+  ];
 
-  # circular dep on webtest
+  # Circular dependency on webtest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/wsgiproxy2/default.nix:26]: This can easily be circumvented

**Pull request merged**